saleor/dashboard/product/views.py

```diff
@@ -2,66 +2,83 @@
 
 from django.contrib import messages
 from django.core.urlresolvers import reverse_lazy
-from django.db import transaction
 from django.shortcuts import get_object_or_404, redirect
 from django.template.response import TemplateResponse
 from django.utils.translation import ugettext_lazy as _
 from django.views.generic import DeleteView
 
-from ...product.models import Product
+from ...product.models import Product, ProductImage, Stock, ProductAttribute
 from ..utils import paginate
 from ..views import StaffMemberOnlyMixin, staff_member_required
-from .forms import (ProductImageFormSet, ProductClassForm, get_product_form,
-                    get_product_cls_by_name, get_variant_formset)
+from .forms import (ProductClassForm, get_product_form,
+                    get_product_cls_by_name, get_variant_form,
+                    ProductImageForm, get_verbose_name, StockForm,
+                    get_variant_cls, ProductAttributeForm,
+                    AttributeChoiceValueFormset, VariantAttributesForm)
 
 
 @staff_member_required
 def product_list(request):
-    products = Product.objects.prefetch_related('images').select_subclasses()
+    products = Product.objects.prefetch_related('images')
     form = ProductClassForm(request.POST or None)
     if form.is_valid():
-        cls_name = form.cleaned_data['cls_name']
-        return redirect('dashboard:product-add', cls_name=cls_name)
+        product_cls = form.cleaned_data['product_cls']
+        return redirect('dashboard:product-add', product_cls=product_cls)
     products, paginator = paginate(products, 30, request.GET.get('page'))
     ctx = {'products': products, 'form': form, 'paginator': paginator}
     return TemplateResponse(request, 'dashboard/product/list.html', ctx)
 
 
 @staff_member_required
-def product_details(request, pk=None, cls_name=None):
-    if pk:
-        product = get_object_or_404(Product.objects.select_subclasses(), pk=pk)
-        title = product.name
+def product_details(request, pk=None, product_cls=None):
+    creating = pk is None
+    initial = {}
+    if creating:
```

Comment:
I consider this to be an anti-pattern. Create a separate `create_product` view and keep the logic clean and separate.